packages/m365_defender/data_stream/event/elasticsearch/ingest_pipeline/pipeline_device.yml

```
@@ -1,1256 +1,1907 @@
-      ignore_missing: true
-  - set:
-      field: host.id
-      copy_from: m365_defender.event.device.id
-      ignore_empty_value: true
-  - append:
-      field: related.hosts
-      value: '{{{host.id}}}'
-      if: ctx.host?.id != null
-      allow_duplicates: false
-  - rename:
-      field: json.properties.InitiatingProcessCommandLine
-      target_field: m365_defender.event.initiating_process.command_line
-      ignore_missing: true
-  - set:
-      field: process.command_line
-      copy_from: m365_defender.event.initiating_process.command_line
-      ignore_empty_value: true
-  - rename:
-      field: json.properties.InitiatingProcessMD5
-      target_field: m365_defender.event.initiating_process.md5
-      ignore_missing: true
-  - set:
-      field: process.hash.md5
-      copy_from: m365_defender.event.initiating_process.md5
-      ignore_empty_value: true
-  - append:
-      field: related.hash
-      value: '{{{process.hash.md5}}}'
-      if: ctx.process?.hash?.md5 != null
-      allow_duplicates: false
   - rename:
-      field: json.properties.InitiatingProcessSHA1
-      target_field: m365_defender.event.initiating_process.sha1
+      field: json.properties.AdditionalFields
+      target_field: m365_defender.event.additional_fields
+      tag: rename_json_properties_AdditionalFields
       ignore_missing: true
-  - set:
-      field: process.hash.sha1
-      copy_from: m365_defender.event.initiating_process.sha1
-      ignore_empty_value: true
-  - append:
-      field: related.hash
-      value: '{{{process.hash.sha1}}}'
-      if: ctx.process?.hash?.sha1 != null
-      allow_duplicates: false
   - rename:
-      field: json.properties.InitiatingProcessSHA256
-      target_field: m365_defender.event.initiating_process.sha256
-      ignore_missing: true
-  - set:
-      field: process.hash.sha256
-      copy_from: m365_defender.event.initiating_process.sha256
-      ignore_empty_value: true
-  - append:
-      field: related.hash
-      value: '{{{process.hash.sha256}}}'
-      if: ctx.process?.hash?.sha256 != null
-      allow_duplicates: false
-  - convert:
-      field: json.properties.InitiatingProcessParentId
-      target_field: m365_defender.event.initiating_process.parent_id
-      if: ctx.json?.properties?.InitiatingProcessParentId != ''
-      type: long
+      field: m365_defender.event.additional_fields.direction
+      target_field: m365_defender.event.network_direction
+      tag: rename_additional_fields_direction
       ignore_missing: true
+      if: ctx.m365_defender?.event?.additional_fields != null && ctx.m365_defender?.event?.additional_fields instanceof Map
+
+  # JSON processors to unpack JSON string fields before renaming them
+  - json:
+      field: json.properties.CrlDistributionPointUrls
+      tag: json_json_properties_CrlDistributionPointUrls
+      if: ctx.json.properties.CrlDistributionPointUrls instanceof String && ctx.json.properties.CrlDistributionPointUrls != ''
```

Is `ctx.json?.properties` always non-null?
```suggestion
      if: ctx.json?.properties?.CrlDistributionPointUrls instanceof String && ctx.json.properties.CrlDistributionPointUrls != ''
```
similar below